docs/extending/data-source.md

```diff
@@ -66,6 +66,52 @@ $zipcode_query = HttpQuery::from_array( [
 
 The goal with design was to provide you with flexibility you need to represent any data source.
 
+## HttpDataSource configuration
+
+### **version**: number (required)
+
+There is no built-in versioning logic, but a version number is required for best practice reasons. Changes to the data source could significantly affect [queries](query.md). Checking the data source version is a sensible defensive practice.
+
+### display_name: string (required)
+
+The display name is used in the UI to identify your data source.
+
+### endpoint: string
+
+This is the default endpoint for the data source and can save repeated use in queries. We would suggest putting the root API URL here and then manipulating it as necessary in individual [queries](query.md).
+
+### request_headers: array
+
+Headers will be set according to the properties of the array. When providing authentication credentials, take care to keep them from appearing in code repositories. We strongly recommend using environment variables or other secure means for storage.
+
+## Additional parameters
+
+You can add any additional parameters that are necessary for your data source. In our [Airtable example](../../example/airtable/events/register.php), you can see that we are setting values for the Airtable `base` and `table`.
+
+Consider adding whatever configuration would be useful to queries. As an example, queries have an `endpoint` property. Our [Zip code example](../../example/rest-api/zip-code/register.php) sets the endpoint with a function:
+
+```php
+$zipcode_query = HttpQuery::from_array( [
+    'data_source' => $zipcode_data_source,
+    'endpoint' => function ( array $input_variables ) use ( $zipcode_data_source ): string {
+        return $zipcode_data_source->get_endpoint() . $input_variables['zip_code'];
+    },
+])
+```
+
+In this case the `data_source` has a built in `get_endpoint()` method. You could also retrieve the endpoint this way:
+
+```php
+$zipcode_query = HttpQuery::from_array( [
+    'data_source' => $zipcode_data_source,
+    'endpoint' => function ( array $input_variables ) use ( $zipcode_data_source ): string {
+        return  $zipcode_data_source->to_array()['service_config']['endpoint'] . $input_variables['zip_code'];
+    },
+])
+```
```

Comment:
I don't think we want to promote this method. Using and promoting the class interface consistently frees us to change internals in the future or add filtering.

```suggestion
```